File: paper_scissors_stone.rb
puts """


#### This is a game of paper - scisors - stone ####
#### Good Luck and, may The Force be with you... ##



""" 

          
def graph(play)  # depending on the player's pick, print the graph

	# for the ASCII representation 
	paper = """                                   
	      ________                                      
	     /       /                                         
	    / paper /                                                          
	   /       /                                                
	  /_______/             
	                                          
	           """                                                
	                    
	scissors = """
	   __   _______            
	  |__| /_//____|            
	   __|<_________                    
	  |___/|__//___/    
	      
	             """
	
	stone = """
	     __                                                             
	  __/   \_
	     |_/_|                                                 
	     \____|                                                   
	   __\___/          
	                                                           
	          """
          
	if play == "pa"
		puts paper
	elsif play == "st"
		puts stone
	elsif play == "sc"
		puts scissors
	end
end


$com = 0  # the score trackers
$you = 0
$round = 0

$board = {"pa" => "paper",    # semantics for printing the right strings
"sc" => "scissors",
"st" => "stone"
}


$winboard = {"pa"=> "st",   # what play beats what play
"sc"=> "pa",
"st"=> "sc"
}

def compick()         # the computer's pick
	n = rand(3)
	if n == 0
		$cpick = "pa" # for paper
	elsif n == 1
		$cpick = "sc" # for scizors
	elsif n == 2
		$cpick = "st" # for stone
	end
end


def result(you, pc) # define winning play
	if you == pc
		puts " "
		puts "It's a draw"
		puts " "
	elsif $winboard[you] == pc
		puts " "
		puts "Apparently you won..."
		$you += 1
	else
		puts " "
		puts "Seems you lost..."
		$com += 1
	end
end


def play()		# pick, and compare with the computer
	puts """	pick a play
	
	pa for paper
	sc for scissors
	st for stone
	
	"""
	pick = gets.chomp   # your pick
	if pick == "pa" or pick == "st" or pick == "sc"
		compick()       # computer's pick
		puts ""
		puts ""
		print "   YOU"   # print the graphics
		graph(pick)   
		print "   COM"
		graph($cpick)
		result(pick, $cpick) # compare and print results
		$round += 1
		puts " "
		puts "In round #{$round}  ->  Score: YOU #{$you.to_s} vs COM #{$com.to_s} " 
		puts " "
	else  # error handling
		puts ""
		puts "you typed something else:"
		print ""
	end
end



# The main game loop
game = true

while game
	puts "Play a round?"
	puts "(y or n)"
	x = gets.chomp()
	if x == "n"
		game = false
		puts "" 
		puts "quiting..."
		print ""
	elsif x == "y"
		play()
	else 
		puts ""
		puts "you typed something else"
		puts""
	end
end

File: paper_scissors_stone.py
import random

print """


#### This is a game of paper - scisors - stone ####
#### Good Luck and have fun! ##



"""

#### globals for keeping track of the score and the round ####

com = 0  # the score trackers
you = 0
rnd = 0  # the round wich we are in
cpick = 0 # this is the computer's pick and is used by many functions


def graph(play):  # depending on the player's pick, print the graph

	# ASCII representations of the possible picks
	paper = """
	      ________
	     /       /
	    / paper /
	   /       /
	  /_______/

	           """

	scissors = """
	   __   _______
	  |__| /_//____\
	   __|<_________
	  |___/|__//___/

	             """

	stone = """
	     __ _
	  __/  |_\
	     |_/__|
	     \____|
	   __\___/

	          """

	if play == "pa":    # for paper
		print paper
	elif play == "st":  # for scizors
		print stone
	elif play == "sc":  # for stone
		print scissors


def compick():        # the computer's pick
	n = random.randint(0,3)
	global cpick
	if n == 0:
		cpick = "pa"
	elif n == 1:
		cpick = "sc"
	elif n == 2:
		cpick = "st"


def result(yplay, pcplay): # define winning play

	global you # for adjusting the score record
	global com

	winboard = {"pa": "st",   # what play beats what play
	"sc": "pa",
	"st": "sc"}

	if yplay == pcplay:
		print "\n It's a draw \n"
	elif winboard[yplay] == pcplay:
		print "\n Apparently you won..."
		you += 1 # adjusting the score record
	else:
		print "\n Seems you lost..."
		com += 1 # adjusting the score record


def play():		# pick, and compare with the computer
	print """	pick a play

	pa for paper
	sc for scissors
	st for stone

	"""
	pick = raw_input()  # your pick
	if pick == "pa" or pick == "st" or pick == "sc":
		compick()       # computer's pick
		print "\n   YOU"   # print the graphics
		graph(pick)
		print "   COM"
		global cpick
		graph(cpick)
		result(pick, cpick) # compare and print results
		global rnd
		rnd += 1  # adjust the round count
		global rnd, you, com
		print "\n In round %d  ->  Score: YOU %d vs COM %d \n" % (rnd, you, com)
	else:  # error handling
		print "\n you typed something else - retype: \n"


# The main game loop
game = True

while game:
	print "Play a round?"
	x = raw_input("(y or n)\n")
	if x == "n":
		game = False
		print "\n quiting... \n"
	elif x == "y":
		play()
	else:
		print "\n you typed something else \n"
